### PR TITLE
Publish -abstractions NuGet packages to GitHub NuGet repository

### PR DESCRIPTION
**Pull request by MaxKsyunz**
1. Updated ci.yml workflow to publish the packages.
2. Required to update RepositoryUrl for all packages so that GitHub accepts the packages.
3. Added a job to delete all but the most recent packages.
4. Packages are deleted and published only to commits to main branch.

### Additional Reviewers
@raymond-lum 
@alexmeizer 
@guiangumpac 
@Yury-Fridlyand

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -5,7 +5,11 @@
     <Authors>OpenSearch Project and contributors</Authors>
     <Copyright>OpenSearch</Copyright>
     <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
-    <RepositoryUrl>https://github.com/opensearch-project/opensearch-net-abstractions</RepositoryUrl>
+    <RepositoryUrl>https://github.com/Bit-Quill/opensearch-net-abstractions-1</RepositoryUrl>
+    <!-- Use the value below when merging to upstream.
```
Comment (raymond-lum) [inline src/Directory.Build.props]: You've obviously tested this. Do we put in the upstream version for fear of missing this change later?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,12 +23,42 @@ jobs:
...
         path: build/output/*
+
+    - run: ./build.sh validatepackages -s true
+      name: Validate *.npkg files that were created
+
+    - name: Publish packages to GitHub package repository
+      if: github.event_name == 'push' && startswith(github.ref, 'refs/heads/main')
+      shell: bash
+      run:  |
+         OWNER=$GITHUB_REPOSITORY_OWNER
```
Comment (raymond-lum) [inline .github/workflows/ci.yml:45]: Is this for readability of the subsequent line?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -28,12 +23,25 @@ jobs:
...
     - uses: actions/upload-artifact@v2
       with:
         name: nuget-packages
         path: build/output/*
+
+    - run: ./build.sh validatepackages -s true
+      name: "validate *.npkg files that were created"
+
+    - name: publish canary packages github package repository
+      #if: github.event_name == 'push' && startswith(github.ref, 'refs/heads')
```
Comment (raymond-lum) [inline .github/workflows/ci.yml]: Is this commented out code needed?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MaxKsyunz) [reply .github/workflows/ci.yml:45]: Exactly. This way it won't need to be updated when we merge to upstream.

File: src/Directory.Build.props (hunk 1 above)
Comment (MaxKsyunz) [reply src/Directory.Build.props]: @guiangumpac  brought up the same point -- I changed it to point to opensearch-project.

**Comment by MaxKsyunz** [comment]
Redoing this PR from a branch with commits that include DCO's.
See https://github.com/Bit-Quill/opensearch-net-abstractions-1/pull/2

**Pull request closed without merging**